source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -0,0 +1,206 @@
+#include "extensions/filters/http/cache/cache_filter.h"
+
+#include "envoy/registry/registry.h"
+
+#include "absl/memory/memory.h"
+#include "absl/strings/string_view.h"
+
+using absl::string_view;
+using absl::WrapUnique;
+using Envoy::Http::FilterDataStatus;
+using Envoy::Http::FilterHeadersStatus;
+using Envoy::Http::HeaderEntry;
+using Envoy::Http::HeaderMap;
+using Envoy::Http::HeaderMapPtr;
+using Envoy::Registry::FactoryRegistry;
+using Envoy::Stats::Scope;
+using std::function;
+using std::make_shared;
+using std::string;
+using std::vector;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace {
+
+bool isCacheableRequest(HeaderMap& headers) {
+  const HeaderEntry* method = headers.Method();
+  // TODO(toddmgreer) Also serve HEAD requests from cache.
+  // TODO(toddmgreer) Check all the other cache-related headers.
+  return ((method != nullptr) && method->value().getStringView() == "GET");
```

Comment:
Headers::get().MethodValues.Get

(see source/common/http/headers.h for a list of these)

Reply:
Done.